src/screens/DiscoveryScreenV2/hooks/useDiscovery.js

```diff
@@ -57,10 +58,13 @@ const useDiscovery = () => {
   };
 
   const topics = React.useMemo(() => {
-    return discovery.initialTopics.map((item) => ({
-      ...item,
-      following: item.following !== undefined ? item.following : item.user_id_follower !== null
-    }));
+    return uniqBy(
+      discovery.initialTopics.map((item) => ({
+        ...item,
+        following: item.following !== undefined ? item.following : item.user_id_follower !== null
+      })),
+      (item) => item.topic_id
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This change ensures that there are no duplicate entries in the `topics` array based on `topic_id`. This is a good approach to prevent redundancy. However, the ternary operator used here might lead to confusion. It would be better to use a more explicit conditional statement for readability.
```diff
- following: item.following !== undefined ? item.following : item.user_id_follower !== null
+ following: item.following !== undefined ? item.following : (item.user_id_follower !== null ? true : false)
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->